src/pages/SignIn/components/UserSignIn.jsx

```diff
@@ -1,42 +1,79 @@
 import { signInWithEmailAndPassword } from 'firebase/auth';
-import React, { useRef } from 'react';
+import React, { useRef, useState } from 'react';
 import { auth } from '../../../utils/firebase';
 import { Input, Button } from '@nextui-org/react';
+import { EyeFilled, EyeInvisibleFilled } from '@ant-design/icons';
 
 const UserSignIn = () => {
   const email = useRef();
   const password = useRef();
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isInvalid, setIsInvalid] = useState(false);
+  const [isVisible, setIsVisible] = React.useState(false);
+
+  const toggleVisibility = () => setIsVisible(!isVisible);
+
   const signIn = (e) => {
     e.preventDefault();
     signInWithEmailAndPassword(auth, email.current.value, password.current.value)
       .then((userCredential) => {
         console.log(userCredential);
       })
       .catch((error) => {
-        console.log(error);
+        console.log(error.code);
+        const message = error.code.replace('-', ' ').replace('auth/', '');
+        console.log('message', message);
+        setErrorMessage(message);
+        setIsInvalid(true);
       });
   };
 
+  const handleInputClick = () => {
+    setIsInvalid(false);
+  };
+
   return (
     <div className="sign-in-container">
       <form id="myForm" onSubmit={signIn}>
         <div className="flex w-[400px] mx-auto flex-col flex-wrap mb-6 md:mb-0 gap-4">
           <Input
+            label="Email"
             size="md"
-            type="email"
             variant="bordered"
-            label="Email"
             isRequired={true}
             ref={email}
+            isInvalid={isInvalid}
```

Comment:
another way i can think of without using isInvalid state and handleInputClick is
```js
const signIn = (e) => {
  e.preventDefault();
  setErrorMessage(''); // instead of handleInputClick

  const formData = new FormData(e.target);
  const email = formData.get('email');
  const password = formData.get('password');

  signInWithEmailAndPassword(auth, email, password)
    .catch((error) => {
      const message = error.code.replace('-', ' ').replace('auth/', '');
      setErrorMessage(message);
    });
};
// ...
<Input
  isRequired
  name="email"
  type="email"
  label="Email"
  size="md"
  variant="bordered"
  isInvalid={errorMessage.length > 0}
  color={errorMessage.length > 0 ? 'danger' : 'default'}
/>
```